Frontend

N1ghtmarex opened the pull request.

N1ghtmarex committed:
Init Frontend project

N1ghtmarex committed:
Добавлен атрибут AllowAnonymous для методов получения локации

N1ghtmarex committed:
Добавлено использование статических файлов

N1ghtmarex committed:
Добавлена папка wwwroot для статических файлов

N1ghtmarex committed:
Изменен путь сохранения изображений

N1ghtmarex committed:
Изменен путь, возвращаемый в результате сохранения файла

N1ghtmarex committed:
- В .gitignore добавлены статические файлы
- Добавлены CORS - AllowAnyOrigin
- Реализованы главная страница со всеми локациями и страница детальной информации о локации

N1ghtmarex committed:
Изменено название в Metadata

N1ghtmarex committed:
Header и Footer вынесены в layout.tsx

N1ghtmarex committed:
- Изменены CORS
- Добавлена страница регистрации

N1ghtmarex committed:
Добавлена интеграция с Keycloak

N1ghtmarex committed:
Добавлен метод получения текущего пользователя

N1ghtmarex committed:
Добавлена проверка роли пользователя

N1ghtmarex committed:
Добавлена страница создания новой локации

N1ghtmarex committed:
Исправлена публикация изображений

N1ghtmarex committed:
Добавлен метод получения типов локаций

N1ghtmarex committed:
Добавлено получение типов локаций с сервера

N1ghtmarex committed:
Добавлена проверка на авторизацию

N1ghtmarex committed:
Выполнен выход из аккаунта если не удалось получить роль

N1ghtmarex committed:
При получении локации возвращается Enum Description

N1ghtmarex committed:
Исправлено добавление новой локации

N1ghtmarex committed:
Исправлено отображение локаций без аватара

N1ghtmarex committed:
Кнопки на карточке локации выровнены по нижнему краю

N1ghtmarex committed:
Добавлено явное указание Keycloak в качестве провайдера аутентификации

N1ghtmarex committed:
Исправлено отображение детальной страницы о локации без аватара

N1ghtmarex committed:
- Добавлено отображение изображений при получении избранных локаций
- Реализован функционал сохранения и удаления локаций из избранного
- Добавлена страница избранных локаций

N1ghtmarex committed:
Добавлены ссылки на регистрацию и авторизацию

N1ghtmarex committed:
Изменен дизайн карточек локаций

N1ghtmarex committed:
fix: Исправлен переход на страницу локации при нажатии на тип локации или кнопку добавить в избранное

N1ghtmarex committed:
fix: обновлено использование иконо для избранного

N1ghtmarex committed:
refactor: Стили для карточки локации вынесены в отдельный scss файл

N1ghtmarex committed:
refactor: - Вынесены стили для главной страницы
- Выполнено подключение глобальных стилей в layout.tsx

v4dikos1 committed:
Merge pull request #3 from N1ghtmarex/UserAuthRegister

User auth register

v4dikos1 commented on Backend/Application/Users/Handlers/UserQueriesHandler.cs:
```diff
@@ -47,5 +50,18 @@ public async Task<PagedResult<UserListViewModel>> Handle(GetUsersListQuery reque
+
+        public async Task<UserViewModel> Handle(GetCurrentUserQuery request, CancellationToken cancellationToken)
+        {
+            var user = await userService.GetUserByExternalIdAsync(httpContext.IdentityUserId, cancellationToken);
```
Я бы еще понял, если нужна была аватарка или еще чего. Но разве эту информацию нельзя достать на фронте из токена?

N1ghtmarex replied:
Токен кейклока не содержит инфу из БД бэкенда

N1ghtmarex replied:
А именно (на данный момент) мне необходимо получить роль, которая хранится в БД, отдельно от кейклока

N1ghtmarex replied:
добавил извлечение ролей из токена

v4dikos1 commented on Frontend/app/favorites/page.tsx:
```diff
@@ -0,0 +1,110 @@
+      setLoading(true);
+
+      const loadFavorites = async () => {
+      try {
```
Может вынести в отдельный интерфейс(сервис) запросы в API? 
Тогда код будет чище

N1ghtmarex replied:
исправил

v4dikos1 commented on Frontend/app/location/[id]/page.tsx:
```diff
@@ -0,0 +1,160 @@
+  const [coordinates, setCoordinates] = useState<[number, number] | null>(null);
+
+  const getLocation = () => {
+    axios.get(`${apiUrl}/api/locations/${id}`)
```
Аналогично, довольно сильно засоряет основной код

N1ghtmarex replied:
исправил

v4dikos1 commented on Frontend/app/page.tsx:
```diff
@@ -0,0 +1,53 @@
+    );
+  }
+
+  const getFavoriteLocations = async () => {
```
аналогично

N1ghtmarex replied:
исправил

v4dikos1 commented on Frontend/components/LocationCard.tsx:
```diff
@@ -0,0 +1,58 @@
+    setIsFavoriteParameter(isFavorite);
+  }, [isFavorite]);
+  
+  const favorite = async (id: string) => {
```
аналогично

N1ghtmarex replied:
исправил

v4dikos1 committed:
Merge pull request #4 from N1ghtmarex/BusinessLogic

Business logic

v4dikos1 committed:
Merge pull request #6 from N1ghtmarex/UserAuthRegister

Merge 'UserAuthRegister' into 'main'

N1ghtmarex committed:
Merge branch 'main' into Frontend

N1ghtmarex committed:
fix: исправлены импорты

N1ghtmarex committed:
refactor: Вынесены запросы к Api
+ Добавлены отдельные сервисы для запросов к Api
+ Добавлена кастомизация axios
* Указан явный маппинг Location.Id в Location-маппере

N1ghtmarex committed:
refactor: Изменено получение локации
+ Добавлен метод получения конкретной локации
+ В модель локации добавлены поля широты и долготы
- Удалена модель адреса

N1ghtmarex committed:
refactor: Получение роли на клиенте
+ В модель Session добавлено поле Roles
+ Теперь роль извлекается из токена
